### DISCUSSION TITLE
Github deleted my entire project. Help?

### DISCUSSION THREAD

**Discussion by ponypaw** [community/community#137213] [New to GitHub]
I was working on a project today and wanted to back it up on github. I use Github Desktop. I opened up Github Desktop and committed the files, then tried to push changes. I got the error "unable to push commits because there are commits on the remote that are not present on your local branch" so I decided to make a new branch to save everything and then work on fixing the main one because I had an issue before that corrupted my project to the point I had to start over. 

When I did this is uploaded one file instead of the hundred or so it should have. I still don't understand why. The program says the folder is missing. When I open the folder manually, it shows the one file and the git ignores that it uploaded. All the rest has been completely removed. It's not in my recycle bin either. 

Is there any way to get it all back? Right now I'm afraid to touch anything for fear of it being gone forever.

**Comment by Yackens** [comment]
To recover your files:

1. Check the .git Folder: Look for stashed changes with git stash list and apply them with git stash apply.
2. Use File Recovery Software: Try tools like Recuva or TestDisk to recover deleted files.
3. Check Git Logs: Use git reflog to see if your files were part of any recent commits.
4. Switch Branches: Check other branches with git checkout <branch_name>.
5. GitHub Desktop: Revert any recent changes directly in GitHub Desktop if the session is still open.
6. Backup: Make a backup of your current directory before making any further changes.

These steps should help you recover your files. If you found this answer useful, please, mark my answer as the correct one to help me obtain GitHub achievement: Galaxy Brain 😄

**Comment by ghostinhershell** [comment]
Hi @ponypaw, I'm really sorry to hear it! The best place to go for answers to account-specific questions like this one is opening a ticket on our [Support page](https://support.github.com/contact). Please open a ticket with them.

**Comment by zohaibAlam840** [comment]
Hi @ponypaw It sounds like a stressful situation, but don't panic—there are a few things you can try to recover your files:

1. Check GitHub Desktop for Changes:
Open GitHub Desktop and switch back to the original branch (before you created the new one).
See if there are any uncommitted changes showing in the “Changes” tab. Sometimes GitHub Desktop might recognize files that haven't been committed yet.
2. Look at the Commit History:
Go to the "History" tab in GitHub Desktop and check if there’s a commit that contains your missing files. If you find them there, you might be able to restore them from a previous commit.
3. Check Your Git Repository Locally:
Manually open the folder on your computer where your Git project is stored. Sometimes the files are still there but might not be showing up correctly in GitHub Desktop.
4. Recover Deleted Files:
Even though they’re not in the recycle bin, Git keeps a history of everything. You can try running a git reflog in the command line to see the changes in your local repository. It might help you find the commit before the files disappear.
5. Check Backup Locations:
If you’ve been using any cloud backup service or version control systems (like Dropbox or Time Machine), check there too.
6. Be Careful Moving Forward:
To avoid overwriting or further deleting anything, avoid pushing more changes or moving files around until you're sure of the state of your project.
If you're not comfortable with Git commands, don't worry—just take it slow and feel free to ask for more guidance. Hopefully, one of these steps will get your files back!

**Comment by ngchthien** [comment]
It sounds incredibly frustrating to lose all your hard work! Let's break down what might have happened and how to potentially recover your project.
Understanding the Issue
"Unable to push commits..." error: This usually happens when changes were made directly on GitHub (website) or another device, and your local repository is out of sync.
Single file upload: This indicates something went wrong during the branch creation and push process. GitHub Desktop might have misunderstood which files to include.
Missing files: Your files are unlikely to be completely gone from your computer unless you specifically deleted them. They're likely hiding somewhere, but we need to figure out where.
Recovery Steps
DON'T PANIC: Resist the urge to make major changes until we've exhausted recovery options.
Check GitHub:
Website History: Go to your repository on the GitHub website, navigate to the file that did upload, and look at its history. You might find older versions of your project there.
Deleted Branches: On GitHub, see if any recently deleted branches exist (they can sometimes be restored).
Search Your Computer:
Recent Files: Check your operating system's recent files/documents list.
Search Everything: Use your computer's search function (like Spotlight on Mac or File Explorer on Windows) to search for file names or project keywords. Don't forget to check hidden folders.
Backups: Do you have any automatic backups running (cloud storage, Time Machine, etc.)? If so, check those immediately.
GitHub Desktop History (Less Likely): While not guaranteed, GitHub Desktop might have some local history:
Repository History: Within GitHub Desktop, see if there's a way to view the history of your local repository.
Hidden Cache: GitHub Desktop might keep a local cache of files. This is less likely, but you could search your computer for folders related to GitHub Desktop and see if any remnants of your project exist.
Future Prevention
Regular Backups: Always have a separate backup system outside of GitHub (cloud storage, external drives).
Smaller Commits: Commit your work in smaller chunks more frequently. This creates more restore points if something goes wrong.
GitHub CLI: For complex scenarios, consider learning basic Git commands on the command line. It gives you more control and understanding of what's happening.
If All Else Fails
If you've exhausted these options and still can't find your files, it might be time to accept the loss and start rebuilding. It's a tough lesson, but this experience highlights the importance of robust backup strategies.

**Comment by DavidS2W** [comment]
If you're using VS Code, there is a very easy way to solve this problem. On VS Code, click on the "timeline" tab at the bottom left. Click the three dots and view past versions of saved files. The code should still be there. Copy the code in the files manually and paste it somewhere. You can't retrieve the deleted files directly but at least you can retrieve the code in them.
